### DISCUSSION TITLE
Proposal: navigate to app plugins within other nav sections, like config or alerting

### DISCUSSION THREAD

**Discussion by mjseaman** [grafana/grafana#46837] [Plugins]
## Problem
Grafana Cloud users have a disjointed, distracting navigation experience.

Grafana Cloud's main nav bar contains app plugins which perform a lot of different actions, like:
- Synthetics
- Getting Started (Integrations and Connections) 
- Cloud Alerting
- (soon) Cloud Auth (access policy config for Grafana Cloud Metrics, Logs, Traces, and Alerts)

This list is growing, and several of these apps do not belong in main nav. Rather, they are less distracting and easier to find within other menu items, like Configuration (Cloud Auth) or Alerts (Cloud Alerting).

## Proposal
Give app developers an option to specify where their plugin should live in Grafana Nav.

If app plugin developers have the option to place app plugins within other nav sections, they can make their apps both more discoverable and less distracting.

The most immediate need is to to put the new [Cloud Auth plugin](https://ops.grafana.net/a/grafana-auth-app) in Grafana's config section.

**Comment by mjseaman** [comment]
@tolzhabayev @torkelo 👀

**Comment by tolzhabayev** [comment]
@mjseaman do you have any proposals on which areas might be interesting for "specify where their plugin should live in Grafana Nav?
- Grafana's config is one named already

P.S. Currently we only have one `addToNav` [property](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/#properties-4) which is adding it to the side menu.

**Reply by torkelo** [reply]
To any main section, Grafana OnCall and Incidents should maybe live in the Alerting section?

**Reply by tolzhabayev** [reply]
And what about ordering, do we want to specify position or just always add to the right?

**Reply by mjseaman** [reply]
Specifying position would be helpful. For example, the Cloud Auth team would want to position their plugin next to API Keys in the config section.

**Comment by lukasztyrala** [comment]
FYI @amy-super

**Comment by mjseaman** [comment]
Hi @tolzhabayev, any recent progress on this topic? CC @cinaglia @VesnaDean and @Rydez

**Comment by amy-super** [comment]
👋  We recently ran a card sorting study to see if there are any red flags for things not making sense in Grafana information architecture - really to just try to get some data on what people think belongs together. The study closed April 8, and @nadinevehling is due to synthesize takeaways from the results week of April 18. I'd appreciate if we can check back in with you all once those takeaways are ready to go.

**Comment by tolzhabayev** [comment]
@mjseaman nothing that can be used yet, we still have to figure out how to do the sub-sub-navigation properly that it would work for the majority of app plugins.

We have the same problem now within the "Connections Console" Working Group. We will try to figure it out together with @zsofiakomaromigrafana, @amy-super and @teddyba 

![SCR-20220426-f4w](https://user-images.githubusercontent.com/580672/165262907-8ede8a78-1889-412c-bf7d-056f825fb9c8.png)

![SCR-20220426-f6b](https://user-images.githubusercontent.com/580672/165262936-e0f39032-4112-4459-aad1-08618c77340a.png)

**Comment by mjseaman** [comment]
Some discussion happening here: https://raintank-corp.slack.com/archives/C01C4K8DETW/p1652886170661839

**Comment by tolzhabayev** [comment]
Issue to track this: https://github.com/grafana/grafana/issues/49788

**Comment by tolzhabayev** [comment]
This is possible since https://github.com/grafana/grafana/pull/55484
